### PR TITLE
RSDK-6564 - Document 'MoveOnMap supports static obstacles'

### PR DESCRIPTION
**Pull request by nfranczak**
We add documentation for a new argument to the `MoveOnMap` method found in the motion service.
The argument we would like to add documentation for is the named `obstacles`.
This argument allows a user to specify static obstacles they wish to be known about at plan time. 
The obstacles which are specified should be in the SLAM coordinate coordinate axes, i.e. the axes which are displayed on the control tab for the SLAM card.

### CODE REVIEW

**Comment by nfranczak** [comment]
context for why the code sample for python should not change:
https://github.com/viamrobotics/viam-python-sdk/pull/539#pullrequestreview-1893983300

File: docs/mobility/motion/_index.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ moved = await motion.move(component_name=gripper_name,
     - If a motion begins with a component already in collision with an obstacle, collisions between that specific component and that obstacle will not be checked.
     - The motion service assumes that obstacles are static.
       If a worldstate obstacle is physically attached to a part of the robot such that it will move with the robot, specify it with _transforms_.
-    - Obstacles are defined by a pose and a [geometry](https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry) with dimensions.
+    - Obstacles are defined by a [(pose)](https://pkg.go.dev/go.viam.com/rdk/spatialmath#Pose) and a [(geometry)](https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry) with dimensions.
```
Comment (nfranczak) [inline docs/mobility/motion/_index.md:179]: I am unsure how to fix 
```
ERROR Link text is too non-descriptive: "https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry)"
```

File: docs/mobility/motion/_index.md [hunk 2]
```diff
@@ -393,6 +393,7 @@ Make sure the [SLAM service](/mobility/slam/) you use supports usage of the foll
   - `plan_deviation_m` [(float)](https://docs.python.org/3/library/functions.html#float): The distance in meters that the machine can deviate from the motion plan.
   - `linear_m_per_sec` [(float)](https://docs.python.org/3/library/functions.html#float): Linear velocity this machine should target when moving.
   - `angular_degs_per_sec` [(float)](https://docs.python.org/3/library/functions.html#float): Angular velocity this machine should target when turning.
+- `obstacles` [Optional[Iterable[Geometry]]](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Geometry): Obstacles, specified in the SLAM frame coordinate system, to be considered when planning the motion of the component.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:396]:
```suggestion
- `obstacles` [Optional\[Iterable\[Geometry\]\]](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Geometry): Obstacles, specified in the SLAM frame coordinate system, to be considered when planning the motion of the component.
```

File: docs/mobility/motion/_index.md [hunk 3]
```diff
@@ -437,6 +438,7 @@ execution_id = await motion.move_on_map(component_name=my_base_resource_name,
     - `PlanDeviationM` [(float64)](https://pkg.go.dev/builtin#float64): The distance in meters that the machine can deviate from the motion plan.
     - `LinearMPerSec` [(float64)](https://pkg.go.dev/builtin#float64): Linear velocity this machine should target when moving.
     - `AngularDegsPerSec` [(float64)](https://pkg.go.dev/builtin#float64): Angular velocity this machine should target when turning.
+  - `Obstacles` [([]spatialmath.Geometry)] (https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry): Obstacles, specified in the SLAM frame coordinate system, to be considered when planning the motion of the component.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:441]:
```suggestion
  - `Obstacles` [(\[\]spatialmath.Geometry)] (https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry): Obstacles, specified in the SLAM frame coordinate system, to be considered when planning the motion of the component.
```

File: docs/mobility/motion/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/mobility/motion/_index.md:179]: I think the issue was on the other links - escaping the brackets should fix it

File: docs/mobility/motion/_index.md (hunk 2 above)
Comment (nfranczak) [reply docs/mobility/motion/_index.md:396]: thanks!

File: docs/mobility/motion/_index.md [hunk 4]
```diff
@@ -437,6 +438,7 @@ execution_id = await motion.move_on_map(component_name=my_base_resource_name,
     - `PlanDeviationM` [(float64)](https://pkg.go.dev/builtin#float64): The distance in meters that the machine can deviate from the motion plan.
     - `LinearMPerSec` [(float64)](https://pkg.go.dev/builtin#float64): Linear velocity this machine should target when moving.
     - `AngularDegsPerSec` [(float64)](https://pkg.go.dev/builtin#float64): Angular velocity this machine should target when turning.
+  - `Obstacles` [(\[\]spatialmath.Geometry)] (https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry): Obstacles, specified in the SLAM frame coordinate system, to be considered when planning the motion of the component.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:441]:
```suggestion
  - `Obstacles` [(\[\]spatialmath.Geometry)](https://pkg.go.dev/go.viam.com/rdk/spatialmath#Geometry): Obstacles, specified in the SLAM frame coordinate system, to be considered when planning the motion of the component.
```

**Comment by npentrel** [comment]
Thank you! this is great!

**Review by npentrel — APPROVED** [review]

**Comment by npentrel** [comment]
There are a large number of tickets relating to changes for MoveOnMap - is there anythign that is now still waiting to be documented? @nfranczak

**Comment by nfranczak** [comment]
> There are a large number of tickets relating to changes for MoveOnMap - is there anything that is now still waiting to be documented?

@npentrel I think this is the last of it :D

**Pull request merged**